movement.cpp

```diff
@@ -0,0 +1,440 @@
+  //Check if the value is correct.
+  if (realAngle > 359)
+  {
+    realAngle -= 360;
+  }
+  //Convert the angle to radians.
+  float gradRadian = (realAngle * pi) / 180;
+  //Variables for the resulting pwm of the function.
+  float pwmM1M3 = 230 * sin(gradRadian);
+  float pwmM2M4 = 230 * cos(gradRadian);
+  //If the pwm is negative, this bools will be used to tell the motor to go backwards.
+  bool orientationM1M3 = 0, orientationM2M4 = 0;
+  
+  //If pwm is negative, the value of the bools changes to true, and the pwm is inverted.
+  if (pwmM1M3 < 0)
+  {
+    orientationM1M3 = true;
+    pwmM1M3 *= -1;
+  }
+  if (pwmM2M4 < 0)
+  {
+    orientationM2M4 = true;
+    pwmM2M4 *= -1;
+  }
+
+  //Since the value is true, the motors will go backwards.
+  if (orientationM1M3)
+  {
+    front_left_motor_.changePwm(pwmM1M3);
+    back_right_motor_.changePwm(pwmM1M3);
+    front_left_motor_.backward();
+    back_right_motor_.backward();
+  }
+  //When the value is false, it will go forward.
+  if (!orientationM1M3)
+  {
+    front_left_motor_.changePwm(pwmM1M3);
+    back_right_motor_.changePwm(pwmM1M3);
+    front_left_motor_.forward();
+    back_right_motor_.forward();
+  }
+  //The same applies for the bool controling de front_right_motor_ and the back_left_motor_.
+  if (orientationM2M4)
+  {
+    front_right_motor_.changePwm(pwmM2M4);
+    back_left_motor_.changePwm(pwmM2M4);
+    front_right_motor_.backward();
+    back_left_motor_.backward();
+  }
+  if (!orientationM2M4)
+  {
+    front_right_motor_.changePwm(pwmM2M4);
+    back_left_motor_.changePwm(pwmM2M4);
+    front_right_motor_.forward();
+    back_left_motor_.forward();
+  }
+}
+
+//Function used to stop the robot.
+void Movements::stop()
+{
+  front_left_motor_.stop();
+  front_right_motor_.stop();
+  back_left_motor_.stop();
+  back_right_motor_.stop();
+}
+
+//Function used to follow the ball.
+void Movements::followBall()
+{
+  /*  
+ *  -1 = Ball wasn't detected
+ *  0 = Possession of ball. This case is used to end the function.
+ *  1 = Ball in the middle of the image. In this case the robot should only advance straight forward.
+ *  2 = Ball in the top left part of the image.
+ *  3 = Ball in the top right part of the image.
+ *  4 = Ball in the bottom left part of the image.
+ *  5 = Ball in the bottom right part of the image.
+ */
+  pixy_.pixyCam(1);
+
+  int caso = -1;
+
+  if (((pixy.pelotaX > 125) && (pixy.pelotaX < 150)) && pixy.pelotaY < 25)
+  {
+    caso = 0;
+  }
+  else if ((pixy.pelotaX > 135) && (pixy.pelotaX < 185))
+  {
+    caso = 1;
+  }
+  else if (pixy.pelotaX > 185)
+  {
+    if (pixy.pelotaY > 120)
+    {
+      caso = 2;
+    }
+    else if (pixy.pelotaY < 120)
+    {
+      caso = 4;
+    }
+  }
+  else if (pixy.pelotaX < 135)
+  {
+    if (pixy.pelotaY > 120)
+    {
+      caso = 3;
+    }
+    else if (pixy.pelotaY < 120)
+    {
+      caso = 5;
+    }
+  }
+
+  switch (caso)
+  {
+
+  case -1:
+    while (pixy.pelotaX == -1)
```

Comment:
In this logic, the robot just moves left n' right until it finds it, right?
Does this work in practice?
Can you think in any case in which these will not work? What did previous teams have made for these?